packages/ui/src/views/webhook/webhook-list/repo-webhook-list.tsx

```diff
@@ -0,0 +1,93 @@
+import { Badge, NoData, StackedList } from '@/components'
+
+import { WebhookType } from './types'
+
+export interface PageProps {
+  webhooks?: WebhookType[]
+  LinkComponent: React.ComponentType<{ to: string; children: React.ReactNode }>
+  handleResetFilters?: () => void
+  hasActiveFilters?: boolean
+  query?: string
+  handleResetQuery: () => void
+}
+
+const Title = ({ title, isEnabled }: { title: string; isEnabled: boolean }) => (
+  <div className="inline-flex items-center gap-2.5">
+    {title}
+    <Badge size="sm" disableHover borderRadius="full" theme={isEnabled ? 'success' : 'muted'}>
+      {isEnabled ? 'Enabled' : 'Disabled'}
+    </Badge>
+  </div>
+)
+
+export function RepoWebhookList({
+  webhooks,
+  LinkComponent,
+  handleResetFilters,
+  hasActiveFilters,
+  query,
+  handleResetQuery
+}: PageProps) {
+  return (
+    <>
+      <StackedList.Root>
+        {webhooks && webhooks.length > 0 ? (
+          <>
+            {webhooks.map((webhook, webhook_idx) => (
+              <LinkComponent key={`${webhook.name}-${webhook_idx}`} to={webhook.name}>
+                <StackedList.Item key={webhook.name} isLast={webhooks.length - 1 === webhook_idx}>
+                  <StackedList.Field
+                    primary
+                    description={webhook.description}
+                    title={<Title title={webhook.name} isEnabled={webhook.enabled} />}
+                    className="gap-1.5"
+                  />
+                  <StackedList.Field
+                    title={
+                      <>
+                        Updated <em>{webhook.createdAt}</em>
+                      </>
+                    }
+                    right
+                    label
+                    secondary
+                  />
+                </StackedList.Item>
+              </LinkComponent>
+            ))}
+          </>
+        ) : (
+          <div className="flex min-h-[50vh] items-center justify-center py-20">
+            {hasActiveFilters || query ? (
+              <NoData
+                iconName="no-search-magnifying-glass"
+                title="No search results"
+                description={['Check your spelling and filter options,', 'or search for a different keyword.']}
+                primaryButton={{
+                  label: 'Clear search',
+                  onClick: handleResetQuery
+                }}
+                secondaryButton={{
+                  label: 'Clear filters',
+                  onClick: handleResetFilters
+                }}
+              />
+            ) : (
+              <NoData
+                iconName="no-data-folder"
+                title="No webhooks yet"
+                description={['There are no webhooks in this project yet.', 'Create new webhooks.']}
+                primaryButton={{
+                  label: 'Create webhook',
+                  onClick: () => {
```

Comment:
the view should just take the handler as a prop and call it.. we should not leave it as an unimplemented inline func.

Reply:
done